test/e2e/datafederation_pe_test.go

```diff
@@ -127,5 +128,26 @@ var _ = Describe("UserLogin", Label("datafederation"), func() {
 			}, df)).To(Succeed())
 			Expect(testData.K8SClient.Delete(testData.Context, df)).Should(Succeed())
 		})
+
+		By("Deleting DataFederation Private Endpoint", func() {
+			_, resp, err := atlasClient.Client.DataFederationApi.DeleteDataFederationPrivateEndpoint(testData.Context,
+				testData.Project.ID(), pe.ID).Execute()
+			Expect(err).To(BeNil())
+			Expect(resp).NotTo(BeNil())
+			Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusNoContent))
+
+			vpe, _, err := atlasClient.Client.DataFederationApi.ListDataFederationPrivateEndpoints(testData.Context,
+				testData.Project.ID()).Execute()
+			Expect(err).To(BeNil())
+			Expect(vpe.GetResults()).To(BeEmpty())
+		})
+		By("Deleting DataFederation Private Endpoint again, as it does not do it the first time around", func() {
+			time.Sleep(5 * time.Second)
+			_, resp, err := atlasClient.Client.DataFederationApi.DeleteDataFederationPrivateEndpoint(testData.Context,
+				testData.Project.ID(), pe.ID).Execute()
+			Expect(err).To(BeNil())
+			Expect(resp).NotTo(BeNil())
+			Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusNoContent))
+		})
```

Comment:
cc @s-urbaniak this does not make sense, but it works. Not sure why.

Still this is not an issue on our side, but behaviour on Atlas or the cloud provider AFAIU.